File: app/src/main/java/com/fionera/demo/activity/RvWithHeaderActivity.java
package com.fionera.demo.activity;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.fionera.base.util.ShowToast;
import com.fionera.demo.R;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

/**
 * @author fionera
 */
public class RvWithHeaderActivity
        extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_rvwith_header);

        ImageView imageView = findViewById(R.id.iv_with_header);

        imageView.setOnClickListener(view -> ShowToast.show("shoot"));

        RecyclerView recyclerView = findViewById(R.id.rv_with_header);
        recyclerView.setLayoutManager(new LinearLayoutManager(this, RecyclerView.VERTICAL, false));
        recyclerView.setAdapter(new RecyclerView.Adapter<MyViewHolder>() {

            @NonNull
            @Override
            public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
                return new MyViewHolder(LayoutInflater.from(RvWithHeaderActivity.this)
                        .inflate(R.layout.rv_recent_session_item, parent, false));
            }

            @Override
            public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {

            }

            @Override
            public int getItemCount() {
                return 10;
            }
        });
    }

    private class MyViewHolder extends RecyclerView.ViewHolder{
        MyViewHolder(View itemView) {
            super(itemView);
        }
    }
}

File: app/src/main/java/com/fionera/demo/activity/RVWithHeaderActivity.kt
package com.fionera.demo.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.fionera.demo.R
import com.fionera.demo.util.ShowToast
import kotlinx.android.synthetic.main.activity_rvwith_header.*

class RVWithHeaderActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_rvwith_header)

        iv_with_header.setOnClickListener({ShowToast.show("shoot")})
        rv_with_header.layoutManager = LinearLayoutManager(this@RVWithHeaderActivity, LinearLayoutManager.VERTICAL, false)
        rv_with_header.adapter = object : RecyclerView.Adapter<MyViewHolder>() {

            override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
                return MyViewHolder(
                        LayoutInflater.from(this@RVWithHeaderActivity).inflate(R.layout.rv_recent_session_item, parent, false))
            }

            override fun onBindViewHolder(holder: MyViewHolder, position: Int) {

            }

            override fun getItemCount(): Int {
                return 10
            }
        }
    }

    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
}
